src/app.js

```diff
@@ -96,40 +81,63 @@ const posts = [
   }
 ];
 
+Object.assign(app.locals, {
+  meta: {
+    title: 'My Blog',
+    description: 'A blog about something awesome!'
+  },
+  header: {
+    title: 'Something Awesome'
+  },
+  footer: {
+    year: new Date().getFullYear()
+  },
+  pages: pages
+});
+
+app.use(middleware.currentPath);
+
 app.get('/', function(req, res) {
 
   res.render('index', {
     posts: posts,
+    req: req
   });
 
 });
 
-app.get('/about', function(req, res) {
+app.get('/posts/:id', function(req, res) {
+  const post = posts.find(function (post) {
+    return post.id === Number(req.params.id)
+  });
 
-  res.render('page', {
-    page: pages.find(function (page){
-      return page.id === 'about';
-    })
-  })
+  res.render('post', {
+    post: post,
```

Comment:
What happens if `post` is undefined? If, for example, I navigate to `/posts/does-not-exist`? I would expect the server to respond with a `404 - Not Found` error.